cmd/dfget/app/root.go

```diff
@@ -234,19 +242,61 @@ func transFilter(filter string) []string {
 	return strings.Split(filter, "&")
 }
 
-func handleNodes() error {
+func handleNodes(nodeWithWeightList []string) ([]string, error) {
 	nodes := make([]string, 0)
 
-	for _, v := range cfg.Nodes {
-		// TODO: check the validity of v.
-		if strings.IndexByte(v, ':') > 0 {
-			nodes = append(nodes, v)
-			continue
+	for _, v := range nodeWithWeightList {
+		node, weight, err := splitNodeAndWight(v)
+		if err != nil {
+			return nil, errors.Wrapf(errortypes.ErrInvalidValue, "node: %s %v", v, err)
+		}
+
+		node, err = handleDefaultPort(node)
+		if err != nil {
+			return nil, errors.Wrapf(errortypes.ErrInvalidValue, "node: %s %v", v, err)
+		}
+
+		for i := 0; i < weight; i++ {
+			nodes = append(nodes, node)
```

Comment:
if  i  input --node 10.10.10.1:8002/20  10.10.10.2:8002/20 ,so they will add 40 times. and at last the dfget will get a random supernode list with length of 80,so does it reasonable

Reply:
so can we do with it like 20/20=1/1 or just ignore it?

Reply:
I will make a consideration. Thanks for your advice.